src/webui/api/torrentscontroller.cpp

```diff
@@ -269,24 +276,50 @@ namespace
 
     QJsonArray getTrackers(const BitTorrent::Torrent *const torrent)
     {
+        const auto toSeconds = [](const std::chrono::nanoseconds &duration) -> qint64
+        {
+            return std::max<qint64>(0, std::chrono::duration_cast<std::chrono::seconds>(duration).count());
+        };
+
+        const auto now = BitTorrent::AnnounceTimePoint::clock::now();
         QJsonArray trackerList;
 
         for (const BitTorrent::TrackerEntryStatus &tracker : asConst(torrent->trackers()))
         {
-            const bool isNotWorking = (tracker.state == BitTorrent::TrackerEndpointState::NotWorking)
-                    || (tracker.state == BitTorrent::TrackerEndpointState::TrackerError)
-                    || (tracker.state == BitTorrent::TrackerEndpointState::Unreachable);
+            QJsonArray endpointsList;
+
+            for (const BitTorrent::TrackerEndpointStatus &endpoint : tracker.endpoints)
+            {
+                endpointsList << QJsonObject
+                {
+                    {KEY_TRACKER_NAME, endpoint.name},
+                    {KEY_TRACKER_UPDATING, endpoint.isUpdating},
+                    {KEY_TRACKER_STATUS, static_cast<int>(endpoint.state)},
+                    {KEY_TRACKER_MSG, endpoint.message},
+                    {KEY_TRACKER_BT_VERSION, static_cast<int>(endpoint.btVersion)},
+                    {KEY_TRACKER_PEERS_COUNT, endpoint.numPeers},
+                    {KEY_TRACKER_SEEDS_COUNT, endpoint.numSeeds},
+                    {KEY_TRACKER_LEECHES_COUNT, endpoint.numLeeches},
+                    {KEY_TRACKER_DOWNLOADED_COUNT, endpoint.numDownloaded},
+                    {KEY_TRACKER_NEXT_ANNOUNCE, toSeconds(endpoint.nextAnnounceTime - now)},
+                    {KEY_TRACKER_MIN_ANNOUNCE, toSeconds(endpoint.minAnnounceTime - now)}
+                };
+            }
+
             trackerList << QJsonObject
             {
                 {KEY_TRACKER_URL, tracker.url},
                 {KEY_TRACKER_TIER, tracker.tier},
                 {KEY_TRACKER_UPDATING, tracker.isUpdating},
-                {KEY_TRACKER_STATUS, static_cast<int>((isNotWorking ? BitTorrent::TrackerEndpointState::NotWorking : tracker.state))},
+                {KEY_TRACKER_STATUS, static_cast<int>(tracker.state)},
```

Comment:
This seems to introduce a breaking change in WebAPI. Existing clients cannot deal with these new states without modifications to support them.